pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -845,7 +845,7 @@ func (b *EndpointBuilder) findServiceWaypoint(endpointIndex *model.EndpointIndex
 		// Currently only ingress will call waypoints
 		return nil, false
 	}
-	if b.service.GetAddressForProxy(b.proxy) == constants.UnspecifiedIP {
+	if !b.service.HasAddressOrAssigned(b.proxy.Metadata.ClusterID) {
```

Comment:
I am wondering could we mutate `GetAddressForProxy` to return AutoAllocatedIP as this function `HasAddressOrAssigned`

Reply:
the problem is that function considers whether DNS auto-allocation/capture is on. However, that only matters for *matching* traffic (listener).Here we are using it to send traffic so we shouldn't consider it

Reply:
Yeah, the dns capture might be the diff.

But here why not fallback to return defaultAddress

Reply:
I think it maybe contained by ClusterVips

Reply:
I think we should probably respect DefaultAddress too. will change it